crates/sargon/src/system/sargon_os/pre_authorization/sargon_os_pre_authorization_status.rs

```diff
@@ -52,15 +52,17 @@ impl SargonOS {
         let mut delay_duration = DELAY_INCREMENT;
 
         loop {
-            // Check if the subinent hasn't expired already
-            if Timestamp::now_utc() > expiration_timestamp {
-                return Err(CommonError::ExpiredSubintent);
-            }
-
             // Increase delay by 1 second on subsequent calls
             delay_duration += DELAY_INCREMENT;
             let sleep_duration = Duration::from_millis(delay_duration);
 
+            // Check if the subinent has already expired. We subtract the `sleep_duration` to consider the case
+            // where the subinent was submitted after last poll, but the `expiration_timestamp` was also reached before this poll.
+            // We must check the subinent status before assuming it had expired.
+            if Timestamp::now_utc().sub(sleep_duration) > expiration_timestamp {
+                return Err(CommonError::ExpiredSubintent);
+            }
+
```

Comment:
so this would allow polling after `Timestamp::now_utc() > expiration_timestamp`, but how would this align with the countdown in UI? Where I would expect that we show the subintent is being expired if `Timestamp::now_utc() >= expiration_timestamp`

Reply:
the countdown in the UI, which only applies for the case of `Expiration::AtTime`, will be displayed before the subintent is signed.

This logic, to poll for its status, will be performed after the subintent is signed.

Reply:
We would show the countdown in the polling slide up also AFAIK